src/components/stackAggregator/index.tsx

```diff
@@ -1,107 +1,115 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {StyleSheet, Animated, Easing, LayoutAnimation} from 'react-native';
+import React, {PureComponent} from 'react';
+import {StyleSheet, Animated, Easing, LayoutAnimation, StyleProp, ViewStyle, LayoutChangeEvent} from 'react-native';
 import {Constants} from '../../helpers';
 import {Colors} from '../../style';
-import {PureBaseComponent} from '../../commons';
-import View from '../view';
+import View, {ViewProps} from '../view';
 import TouchableOpacity from '../touchableOpacity';
-import Button from '../button';
+import Button, {ButtonSize, ButtonProps} from '../button';
 import Card from '../card';
+import {asBaseComponent} from '../../commons/new';
 
 const PEEP = 8;
 const DURATION = 300;
 const MARGIN_BOTTOM = 24;
 const buttonStartValue = 0.8;
 const icon = require('./assets/arrow-down.png');
 
-/**
- * @description: Stack aggregator component
- * @modifiers: margin, padding
- * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/StackAggregatorScreen.js
- */
-export default class StackAggregator extends PureBaseComponent {
-  static displayName = 'StackAggregator';
-
-  static propTypes = {
-    /**
+export type StackAggregatorProps = ViewProps & {
+   /**
      * The initial state of the stack
      */
-    collapsed: PropTypes.bool,
+    collapsed: boolean;
+    /**
+     * Component Children
+     */
+     children: JSX.Element | JSX.Element[]
     /**
      * The container style
      */
-    containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    containerStyle?: StyleProp<ViewStyle>;
     /**
      * The content container style
      */
-    contentContainerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    contentContainerStyle?: StyleProp<ViewStyle>;
     /**
      * The items border radius
      */
-    itemBorderRadius: PropTypes.number,
+    itemBorderRadius?: number;
     /**
      * Props passed to the 'show less' button
      */
-    buttonProps: PropTypes.object,
+    buttonProps?: ButtonProps;
     /**
      * A callback for item press
      */
-    onItemPress: PropTypes.func,
+    onItemPress?: (index: number) => void;
     /**
      * A callback for collapse state will change (value is future collapsed state)
      */
-    onCollapseWillChange: PropTypes.func,
+    onCollapseWillChange?: (change: boolean) => void;
```

Comment:
'changed' instead of 'change' (as 'change' refers to the change itself and not its state)